### DISCUSSION TITLE
[V6] Yup validation for NumberInput value

### DISCUSSION THREAD

**Discussion by 7iomka** [mantinedev/mantine#3941] [Q&A]
In v6 [NumberInput](https://mantine.dev/core/number-input/) component types for value, defaultValue and onChange props were changed. It now expects value to be number | '' (number or empty string) instead of number | undefined. This change was made to address multiple bugs that happened because it was not possible to differentiate controlled and uncontrolled NumberInput.

I need to validate NumberInput by myself with yup schema.
I have 2 cases:

1.  NumberInput is required

2.  NumberInput is optional

Can you tell me please how to validate it for this 2 cases?

The base approach is:
```
value: yup.number()[isRequired ? 'required' : 'optional']()
```

For 2nd case I get this error after blur event happens on an empty NumberInput
```
value must be a `number` type, but the final value was: `NaN` (cast from the value `""`).
```

All api endpoints that requires or returns me an optional number value refer to `number | undefined` type. So I need manually in each case to do something like `value === '' ? undefined : value`, but this is not difficult, but to make a validation schema is much harder.

How do you do it in your projects?

**Comment by rtivital** [comment]
> value === '' ? undefined : value

I do it the same way, there are not much alternatives. If you use `@mantine/form` you can do it in `transformValues`
